am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -45,21 +52,34 @@ public void createResourceAccess(ExplicitAccessRecord explicitAccessRecord) {
     }
 
     /**
-     * Returns `resourceJson` when record with userId and resourceId exist and has READ permissions, otherwise null.
+     * Returns {@link FilterResourceResponse} when record with userId and resourceId exist and has READ permissions,
+     * otherwise null.
      *
      * @param userId       (accessorId)
      * @param resourceId   resource id
      * @param resourceJson json
      * @return resourceJson or null
      */
-    public JsonNode filterResource(String userId, String resourceId, JsonNode resourceJson) {
+
+    public FilterResourceResponse filterResource(String userId, String resourceId, JsonNode resourceJson) {
         AccessManagement explicitAccess = jdbi.withExtension(AccessManagementRepository.class,
             dao -> dao.getExplicitAccess(userId, resourceId));
 
         if (explicitAccess == null) {
             return null;
         }
 
-        return Permissions.hasPermissionTo(explicitAccess.getPermissions(), Permissions.READ) ? resourceJson : null;
+        if (READ.isGranted(explicitAccess.getPermissions())) {
+            Map<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
+            attributePermissions.put("/", Permissions.fromSumOf(explicitAccess.getPermissions()));
```

Comment:
@geek-soft - do we want to use Json pointer?

Reply:
@yarpo for that story yes. This sprint we will make decision which one to use based on CCD input and change that one line if needed.